dd-client/io_runner.sh

```diff
@@ -1,27 +1,31 @@
 #!/bin/bash
 
-#######################################################################################################################
-# Script Name   : io_runner.sh         									      		
-# Description   : Run dd profiles on the /datadir. 
-# Creation Data : 12/10/2018                                                                                          
-# Modifications : None											               		
-# Script Author : Sudarshan					      
-#######################################################################################################################
-
-TEST_SIZE="512k"
-TEST_COUNT="1000"
+blockSize=$(echo $BLOCK_SIZE)
+count=$(echo $COUNT)
+namespace=$(echo $NAMESPACE)
+mountPoint=$(echo $MOUNT_POINT)
+app_label=$(echo $APP_LABEL)
+retry_duration=$(echo $RETRY_DURATION)
+retry_count=$(echo $RETRY_COUNT)
 
 #Verify that the datadir used by the templates is mounted
-if ! df -h -P | grep -q datadir > /dev/null 2>&1; then
-    echo -e "datadir not mounted successfully, exiting \n"
-    exit 1
-fi
+count=0
 
 # Start dd I/O 
-for i in {1..10}
-do
-   echo -e "\nRunning dd profile test with size=$TEST_SIZE ... Wait for results !!\n"
-   dd if=/dev/urandom of=/datadir/f$i bs=$TEST_SIZE count=$TEST_COUNT oflag=dsync 
-done  
-sleep 10
-rm -rf /datadir/*
+while true
+do 
+   podName=$(kubectl get pod -n $namespace -l $app_label --no-headers | grep Running | awk '{print $1}' | head -1)
+   if [ -z "$podName" ]; then
+      echo "Unable to get pod Name. Retrying..."
+      count=$((count +1))
+      sleep $retry_duration
+      if [ "$count" -gt "$retry_count" ]; then
+         exit 1;
+      fi
+   else
+   echo "Writing data on $mountPoint"
+   randomString=$(echo $RANDOM | tr '[0-9]' '[a-z]')
+   kubectl exec -it $podName -n $namespace -- sh -c "cd $mountPoint && dd if=/dev/urandom of=test.$randomString bs=$blockSize count=$count"  
```

Comment:
From this alpine pod - we plan to run a dd command on the application pod (which, I suppose is mostly busubox etc..,? ) With `-it` we seem to be entering the terminal of the remote app/pod -- can we do it w/o that.

Reply:
Also, this implicitly needs dd to be available on the target apps - which I suppose "most have". Base images such as `scratch` do not allow a shell at all -- so do you think a README to go with this (on what this does and req on the app side) can help -- I suppose this is not a new review comment -- we have had dd client from before, but would be good to add.